### PR TITLE
Bugfix where training cost shows as $0.00

### PR DESCRIPTION
**Pull request by Kovbo**
The calculation issue arises because we recalculate tokens for each dataset entry as a background process in datasetEntries.createFromLoggedCalls function:
`countDatasetEntryTokens.enqueue()`

A simple solution is to check if any entries with `null` tokens exist.

If we consider adding more calculations to the dataset creation function, we could add a column to the dataset table that tracks if the dataset is processed.

### CODE REVIEW

**Review by arcticfly — COMMENTED** [review]
Thanks for catching this. I think we should poll while calculations are in progress, right? You can find an example of how we do that in `EvaluationTable.tsx`.

**Comment by Kovbo** [comment]
> Thanks for catching this. I think we should poll while calculations are in progress, right? You can find an example of how we do that in `EvaluationTable.tsx`.

Nice, I thought we did not have an easy way to implement it. Added.

File: app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx [hunk 1]
```diff
@@ -105,6 +105,7 @@ const FineTuneModal = ({ disclosure }: { disclosure: UseDisclosureReturn }) => {
   const [trainingConfigOverrides, setTrainingConfigOverrides] = useState<
     Partial<AxolotlConfig> | undefined
   >();
+  const [calculationRefetchInterval, setCalculationRefetchInterval] = useState(0);
```
Comment (corbt) [inline app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx:108]: Think this could be replaced more elegantly with a `useMemo`.

File: app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx [hunk 2]
```diff
@@ -327,78 +338,80 @@ const FineTuneModal = ({ disclosure }: { disclosure: UseDisclosureReturn }) => {
...
-                      </NumberInputStepper>
-                    </NumberInput>
-                  </VStack>
-                  <HStack>
-                    <Text fontWeight="bold">Number of Epochs</Text>{" "}
-                    <InfoCircle tooltipText="The total number of times the entire dataset is passed forward and backward through a neural network during training." />
-                  </HStack>
+                      <NumberInput
+                        step={0.0001}
+                        min={0.0001}
```
Comment (corbt) [inline app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx]: The minimum here should be 0 actually. It's possible someone will want to train with a learning rate several orders of magnitude smaller than our default.

File: app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx (hunk 1 above)
Comment (arcticfly) [reply app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx:108]: Can it? The data we rely on to determine whether we should be polling or not is fetched from useDatasetTrainingCost, which is also the place where we need to specify our polling interval. I think a combination of useEffect and setState is the best way to get around this circular dependency, but I could be wrong.

Comment (Kovbo) [reply app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx:108]: I'm not sure I understand how to do it.
If I use `useMemo` to handle the state, it will depend on the query response: `price.data?.calculating`. The query response, at the same time, will depend on the interval from `useMemo` state, causing the "variable used before its declaration" error.
Maybe I misunderstood the comment?

Comment (corbt) [reply app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx:108]: Ah I see, yes that makes sense.

File: app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx (hunk 2 above)
Comment (Kovbo) [reply app/src/components/datasets/DatasetContentTabs/General/FineTuneButton.tsx]: Done!

**Review by corbt — APPROVED** [review]
